src/unix/poll.c

```diff
@@ -365,10 +365,7 @@ static epoll_blocked alloc_epoll_blocked(epoll e)
 
 static void check_fdesc(epollfd efd, fdesc f, thread t)
 {
-    /* if edge-triggered, only notify the changes to events */
     u32 events = apply(f->events, t);
-    if (efd->eventmask & EPOLLET)
-        events &= ~efd->lastevents;
```

Comment:
I think this currently breaks some epoll_wait behavior in other circumstances based on the hacky epoll test I have. I will write up some simplified test cases to help identify those areas. I think this particular change is right, but that we'll need some additional changes to not break existing behavior.

Reply:
@sanderssj I modified the eventfd edge handler according to your suggestions, and now it mirrors the Linux behavior. I also added your eventfd-related tests to the epoll runtime tests. Thanks!